core/src/test/java/org/elasticsearch/action/delete/DeleteResponseTests.java

```diff
@@ -0,0 +1,169 @@
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.RandomObjects;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.common.xcontent.XContentHelper.toXContent;
+
+public class DeleteResponseTests extends ESTestCase {
+
+    public void testToXContent() throws IOException {
+        {
+            DeleteResponse response = new DeleteResponse(new ShardId("index", "index_uuid", 0), "type", "id", 3, 5, true);
+            String output = Strings.toString(response);
+            assertEquals("{\"found\":true,\"_index\":\"index\",\"_type\":\"type\",\"_id\":\"id\",\"_version\":5,\"result\":\"deleted\"," +
+                "\"_shards\":null,\"_seq_no\":3}", output);
+        }
+        {
+            DeleteResponse response = new DeleteResponse(new ShardId("index", "index_uuid", 0), "type", "id", -1, 7, true);
+            response.setForcedRefresh(true);
+            response.setShardInfo(new ReplicationResponse.ShardInfo(10, 5));
+            String output = Strings.toString(response);
+            assertEquals("{\"found\":true,\"_index\":\"index\",\"_type\":\"type\",\"_id\":\"id\",\"_version\":7,\"result\":\"deleted\"," +
+                "\"forced_refresh\":true,\"_shards\":{\"total\":10,\"successful\":5,\"failed\":0}}", output);
+        }
+    }
+
+    public void testToAndFromXContent() throws IOException {
+        final XContentType xContentType = randomFrom(XContentType.values());
+
+        // Create a random DeleteResponse and converts it to XContent in bytes
+        DeleteResponse response = randomDeleteResponse();
+        BytesReference responseBytes = toXContent(response, xContentType);
+
+        // Parse the XContent bytes to obtain a parsed
+        DeleteResponse parsedDeleteResponse;
+        try (XContentParser parser = createParser(xContentType.xContent(), responseBytes)) {
+            parsedDeleteResponse = DeleteResponse.fromXContent(parser);
+            assertNull(parser.nextToken());
+        }
+
+        // We can't use equals() to compare the original and the parsed index response
+        // because the random index response can contain shard failures with exceptions,
+        // and those exceptions are not parsed back with the same types.
+
+        // Print the parsed object out and test that the output is the same as the original output
+        BytesReference parsedDeleteResponseBytes = toXContent(parsedDeleteResponse, xContentType);
+        try (XContentParser parser = createParser(xContentType.xContent(), parsedDeleteResponseBytes)) {
+            assertDeleteResponse(response, parser.map());
+        }
+    }
+
+    private static void assertDeleteResponse(DeleteResponse expected, Map<String, Object> actual) {
+        assertEquals(expected.getIndex(), actual.get("_index"));
```

Comment:
I'm about to change this in #22586 so that DeleteResponse/IndexResponse/UpdateResponse don't have to repeat all these checks. There will be a assertDocWriteResponse() method, and here we only have to check for UpdateResponse specific fields.